src/storage/src/hummock/write_limiter.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use arc_swap::ArcSwap;
+use risingwave_common::catalog::TableId;
+use risingwave_hummock_sdk::CompactionGroupId;
+use risingwave_pb::hummock::write_limits::WriteLimit;
+
+pub type WriteLimiterRef = Arc<WriteLimiter>;
+
+#[derive(Default)]
+pub struct WriteLimiter {
+    limits: ArcSwap<HashMap<CompactionGroupId, WriteLimit>>,
```

Comment:
How about build a table-group index to avoid iterator all group for every table-id?